server/controllers/users.js

```diff
@@ -0,0 +1,322 @@
+import { Users, Roles, Documents } from '../models';
+import {
+  passwordHash,
+  authenticateUser,
+  catchError,
+  returnValidationErrors,
+  isRegisteredUser,
+} from '../helpers/utils';
+
+/**
+ * @description creates a user
+ * @function createUser
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const createUser = (req, res) => {
+  req.checkBody('fullname', 'Full name cannot be empty').notEmpty();
+  req.checkBody('email', 'Email cannot be empty').notEmpty();
+  req.checkBody('email', 'Enter a valid email address').isEmail();
+  req.checkBody('password', 'Password cannot be empty').notEmpty();
+  req.checkBody('password', 'Password should be a minimum of 6 characters').notEmpty();
+
+  returnValidationErrors(req, res);
+
+  /** check if provided email already exists user database */
+  isRegisteredUser(req.body.email, 'email')
+    .then((registrationState) => {
+      if (registrationState === false) {
+        Users.create({
+          fullname: req.body.fullname,
+          password: passwordHash(req.body.password),
+          email: req.body.email,
+          roleId: 2,
+        })
+        .then(newUser => res.status(201).send({
+          status: 'ok',
+          userDetails: {
+            userId: newUser.userId,
+            fullname: newUser.fullname,
+            email: newUser.email,
+            roleId: newUser.roleId,
+            created: newUser.createdAt,
+          },
+          message: 'Account creation was successful',
+        }))
+        .catch(() => catchError(res));
+      }
+      return res.status(400).send({
+        status: 'error',
+        message: 'an account with this email already exists',
+      });
+    }
+  );
+};
+
+/**
+ * @description view all user details
+ * @function viewUser
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object of all users
+ */
+const viewUser = (req, res) => {
+  /** set default values for offset and limit */
+  let offset = 0, limit = 20;
+  if (req.query.limit || req.query.offset) {
```

Comment:
Same comment about handling for when only one of these query parameters gets passed applies here too.